update forecasting documentation (#1719)
FlexMeasures/flexmeasures: documentation/features/forecasting.rst

Reviewer: ```suggestion
Of course, the nicest forecasts are the one you don't have to make yourself (it's not an easy field), so do use price or usage forecasts from third parties if available, and load them into FleeMeasures.
```

Author: applied.

> If you need to make your own predictions, forecasting algorithms can be used within FlexMeasures, for instance to assess the expected profile of future consumption/production.

Reviewer: ```suggestion
If you need to make your own predictions, forecasting algorithms can be used within FlexMeasures, for instance to arrive at an expected profile of future solar power production at the site.

FlexMeasures provides a CLI command to generate forecasts (see below). An API endpoint will follow soon.
```

FlexMeasures provides a **fixed-view forecasting infrastructure**.  

> This means that the model is trained once on a historical period and then produces predictions for a future period in one go.  

Reviewer: ```suggestion
This means that from one (fixe) point in time, we forecast a range of events into the future (e.g. 24 hourly events for a span of one day). While the first forecast (one hour ahead) has a small horizon (1H), the last one has a large horizon (24H) and the accuracy between the two will usually differ (it is easier to forecast small horizons).   
```

> At the same time, the design is inspired by **rolling forecasts**, as training and prediction can be repeated in **cycles** until a user-specified end date is reached.  

Reviewer: ```suggestion
At the same time, the design we implemented in FlexMeasures is inspired by **rolling forecasts**, as training and prediction can be repeated in **cycles** until a user-specified end date is reached.  If you ask FlexMeasures for a fixed-point forecast (one cycle), the model is trained once on the most recent applicable historical period and then produces predictions for the requestedfuture period in one go.   
```

This is controlled by the ``forecast_frequency`` parameter, which specifies how often predictions are generated during the forecast period.

> Cycle Example

Reviewer: ```suggestion
How a forecasting cycle works
```

-------------

A single forecasting cycle consists of the following steps:

1. **Training**: Fit the model on a historical window defined by ``train_start`` and ``train_end``.  
2. **Prediction**: Produce forecasts for a horizon defined by ``predict_start`` and ``predict_end``.  
3. **Repeat**: If the global ``end_date`` is not yet reached, move the prediction window forward by ``forecast_frequency`` and repeat steps 1 and 2.

This way, forecasts can cover long ranges while still being based on updated training data in each cycle.

> CLI Parameters

Reviewer: ```suggestion
CLI Command
```

Under this, I would first introduce the CLI command, *then* go into the parameters.

Author: done.

.. code-block:: bash

    flexmeasures add forecasts --from-date 2024-02-02 --to-date 2024-02-02 --horizon 6 --sensor 12  --as-job  # add train-start

Sensor 12 would represent the power readings of your solar power, and here you ask for forecasts for one day (2 February, 2024), with a forecast of 6 hours.

The ``--as-job`` parameter is optional. If given, the computation becomes a job which a worker needs to pick up. There is some more information at :ref:`how_queue_forecasting`.

Fixed-point vs rolling
----------------------

> Unlike previous rolling forecasts, the new infrastructure generates **fixed-point forecasts**:

Reviewer: I believe you explained this well already. We could explain what rolling forecasts are here (using the old explanation about sharing the same horizon). It is not that crucial, just a clarification.

- One reference timestamp.
- Predictions are made for multiple future horizons from that point.
- Periodic retraining ensures forecasts remain accurate.

Regressors
-------------

> If you want to take regressors into account, in addition to merely past measurements (e.g. weather forecasts, see above),

Reviewer: Is this still true?

Author: if you referring to "currently FlexMeasures supports only weather correlations." no this is not true. since our new forecasting infra doesn't look into the closest weather sensor and use as a regressor. we only use explicitly
 passed sensor id's as regressors. but now i think about it should we use that attribute in new forecasting infra?
